### ISSUE TITLE
SECURE flag no longer valid for MySQL external dictionary DDL query

### ISSUE THREAD

**Issue by mmolinac** [ClickHouse/ClickHouse#39242]
**Describe what's wrong**

We use external MySQL dictionaries created by DDL queries like this one:

```SQL
CREATE DICTIONARY common.countries
(
    `id` UInt64,
    `code` String,
    `name` String
)
PRIMARY KEY id
SOURCE(MYSQL(PORT '3306' USER 'staging' PASSWORD 'xxxxxxxxxxxxx' REPLICA (HOST 'staging-db' PRIORITY 1) DB 'staging_v4' WHERE 'type="Country"' TABLE places SECURE 0))
LIFETIME(MIN 3600 MAX 5400)
LAYOUT(HASHED())
```

Depending on the environment we either set `SECURE` to 0 or 1. It is working in our current ClickHouse version and with setting set to 0 on Staging and 1 in Production environments.

That was with ClickHouse version 21.8.5.7, which is our current production version.

With versions 22.3.6.5 onwards, after creating the dictionary and trying to access it, we receive:

```
SQL Error [36] [07000]: Code: 36. DB::Exception: Unexpected key `secure` in dictionary source configuration. (BAD_ARGUMENTS) (version 22.3.8.39 (official build))
, server ClickHouseNode(addr=http:127.0.0.1:14207, db=default)@-1765255297
```

Versions tested:
- 22.3.6.5
- 22.3.8.39
- 22.6.3.35

**Does it reproduce on recent release?**

Yes. Versions tested:
- 22.3.6.5
- 22.3.8.39
- 22.6.3.35

**Enable crash reporting**

We've reproduced the issue with crash reporting enabled.

**How to reproduce**

* ClickHouse server version to use: 22.3.6.5
* Which interface to use: Doesn't matter. Command line will be fine.
* Non-default settings: N/A
* Statements involved:
```SQL
CREATE DICTIONARY common.countries
(
    `id` UInt64,
    `code` String,
    `name` String
)
PRIMARY KEY id
SOURCE(MYSQL(PORT '3306' USER 'staging' PASSWORD 'xxxxxxxxxxxxx' REPLICA (HOST 'staging-db' PRIORITY 1) DB 'staging_v4' WHERE 'type="Country"' TABLE places SECURE 0))
LIFETIME(MIN 3600 MAX 5400)
LAYOUT(HASHED())
```

* Sample data for all these tables: N/A
* Queries to run that lead to unexpected result: Any subsequent query to the external dictionary will provoke the error.

**Expected behavior**

We expect `SECURE` flag to work without giving errors, as it is documented, with versions newer than 21.8.5.7 .

**Error message and/or stacktrace**

```
SQL Error [36] [07000]: Code: 36. DB::Exception: Unexpected key `secure` in dictionary source configuration. (BAD_ARGUMENTS) (version 22.3.8.39 (official build))
, server ClickHouseNode(addr=http:127.0.0.1:14207, db=default)@-1765255297
```

**Additional context**

It does not matter if you use `SECURE` with value 0 or 1, it fails anyway.

With latest versions, it only works if you omit `SECURE` flag.

**Comment by nickitat** [comment]
@mmolinac  I don't see this option in docs. are you sure, that ssl/not ssl connections are used depending on this settings in 21.8?

**Comment by mmolinac** [comment]
@nickitat We started using it on that version in order to being able to use SSL between both ends of the connection. I've just checked that version 21.8.5.7 is able to auto negotiate SSL for MySQL dictionaries (correct me if I'm wrong).
In our Production MySQL DB we only allow SSL communication for the dictionaries user.
If the SSL feature works like this, in an auto-negotiated way, it would be nice to have it in the docs.
I'll check now the behaviour of all other versions I mentioned in the ticket.

**Comment by mmolinac** [comment]
@nickitat thanks to your comment, I tested the creation of MySQL dictionary omitting `SECURE` parameter in all the ClickHouse versions mentioned in the issue. It works in all of them.
What mislead us when we adopted it was the fact that, by providing the parameter, ClickHouse 21.8.5.7 does not complain about the additional parameter. However, subsequent versions do.
That leaves us with maybe a bug in 21.8.5.7 that doesn't complain on non-valid parameter.
FWIW, the newer versions work as we expect.
